Hyperlink `#123` to the respective issue/ticket

muhqu opened the pull request:
Adding this pull request in response to Issue #187. 

I added this a very long time ago to another fork of GitX. Now I rebased this commit onto GitX-dev master.

Here is how it works, just add `gitx.ticketurl` to your repositorys `.git/config`.

```
[gitx]
   ticketurl = "http://trac.example.org/ticket/{id}"
```

To point to github issues just use:

```
$ git config gitx.ticketurl 'https://github.com/rowanj/gitx/issues/{id}'
```

---

ToDo:
- [x] change pattern matching to: replace `{id}` or just append
- [x] auto detect GitHub "origin" for a default
- [x] always make  `#123` links, if not configured show message how to do it
- [x] change `[PBGitHistoryController showConfigureGitxTicketUrl]` to not only show a hint, instead let the user configure the Ticket URL right away... 

![configure-ticket-url](https://f.cloud.github.com/assets/26716/782779/7edc7a4c-ea41-11e2-924e-cff409a478d7.png)

muhqu committed:
replace ticket numbers with links via .git/config configurable gitx.ticketurl
```
[gitx]
   ticketurl = "http://example.org/ticket/*"
```

rowanj commented:
Thanks! I'll look at merging this soon.

Just some notes for myself when I look at this:
- It should be quite simple to add a Preferences option to add this
- Perhaps the match pattern also needs to be configurable
- Auto-detect and default to GitHub pattern if a remote has a GitHub URL

muhqu commented:
I don't think the pattern needs to be configurable, but the pattern should be dead simple. I know replacing `*` with the issue id is not that obvious. I'll change it to look for a placeholder like `{id}` to replace it.If the placeholder is not found the id will just be appended.

Auto-detection for GitHub pattern would be great. But I think it should be sufficient to just check the remote named "origin". 

A preference option doesn't make much sense here, as it is a per-repository setting. Does GitX already allow to modify git-config settings?

If no `gitx.ticketurl` is configured auto-detection failed the numbers could still be linked. When clicked, they could open a popup to configure `gitx.ticketurl` (something like the one used for tag info). Maybe right-click context-menu on a `#123` link could list an entry like "Configure URL..".

muhqu committed:
change pattern matching to: replace `{id}` or just append

rowanj commented:
The preference can be stored/read from the repository (i.e. non-global `git config`) rather than the user defaults; there are a few other things I'd like to store against the repo; and this may be a good excuse for a repo inspector/settings window.

muhqu committed:
auto detect GitHub `remote.origin` as a fallback

muhqu committed:
update gitx.ticketurl scheme in comment

muhqu committed:
always make `#123` links and show a 'Configure URL' context menu which shows a help message on how to configure gitx.ticketurl.

muhqu committed:
implemented TicketURL configuration sheet.
and..
- moved logic away from javascript
- added a "Copy URL" context menu item

muhqu commented:
@rowanj I added a custom view `PBGitXConfigureTicketURLSheet` that lets the user set the `gitx.ticketurl` config setting. maybe some methods are not properly named because I have not much objective-c knowledge... bear with me. ;-)
